subxt/examples/legacy_reconnecting_rpc_client.rs

```diff
@@ -0,0 +1,150 @@
+//! Example to utilize the `reconnecting rpc client` in subxt
+//! which hidden behind behind `--feature unstable-reconnecting-rpc-client`
+//!
+//! To utilize full logs from the RPC client use:
+//! `RUST_LOG="jsonrpsee=trace,reconnecting_jsonrpsee_ws_client=trace"`
+
+#![allow(missing_docs)]
+
+use std::sync::Arc;
+use std::time::Duration;
+
+use subxt::backend::legacy::LegacyBackend;
+use subxt::backend::rpc::reconnecting_rpc_client::{Client, ExponentialBackoff};
+use subxt::backend::rpc::RpcClient;
+use subxt::error::Error;
+use subxt::tx::TxStatus;
+use subxt::{OnlineClient, PolkadotConfig};
+use subxt_signer::sr25519::dev;
+
+// Generate an interface that we can use from the node's metadata.
+#[subxt::subxt(runtime_metadata_path = "../artifacts/polkadot_metadata_small.scale")]
+pub mod polkadot {}
+
+#[tokio::main]
+async fn main() -> Result<(), Box<dyn std::error::Error>> {
```

Comment:
I like this example in that everything is "the same" except it'll retry in the background and you can monitor the reconnects if you like (I assume that bit is entirely optional and I'd probably tweak the comment above that `tokio::spawn` to prefix with "Optional: " but that's just a nit :))

Reply:
I have merged these into a single example with some comments have to use the unstable backend